test/pyapi.jl

```diff
@@ -2,50 +2,56 @@
+            # print(api.read_distribution("parameter-read", "example-samples"))  # expected to fail
 
-            @test read_sample(api, "parameter", "example-estimate") == 1.0
-            @test read_sample(api, "parameter", "example-distribution") isa Real
-            @test read_sample(api, "parameter", "example-samples") ∈ [1, 2, 3]
+            @test read_sample(api, "parameter-read", "example-estimate") == 1.0
+            @test read_sample(api, "parameter-read", "example-distribution") isa Real
+            @test read_sample(api, "parameter-read", "example-samples") ∈ [1, 2, 3]
 
             expected_table = DataFrame(:a => [1, 2], :b => [3, 4])
             expected_array = (data=[1, 2, 3], dimensions=nothing, units=nothing)
-            @test read_table(api, "object", "example-table") == expected_table
-            @test read_array(api, "object", "example-array") == expected_array
+            @test read_table(api, "object-read", "example-table") == expected_table
+            @test read_array(api, "object-read", "example-array") == expected_array
         end
 
         @testset "Write API" begin
-            write_estimate(api, "parameter", "example-estimate", 99.0)
-            @test read_estimate(api, "parameter", "example-estimate") == 99.0
+            # Can't read it before we've written it
+            @test_throws Exception read_estimate(api, "parameter-write", "example-estimate")
+            write_estimate(api, "parameter-write", "example-estimate", 99.0)
+            @test read_estimate(api, "parameter-write", "example-estimate") == 99.0
 
+            @test_throws Exception read_estimate(api, "parameter-write", "example-distribution")
             @testset "Write $d" for d in (Gamma(4, 5), Normal(4, 5))
-                write_distribution(api, "parameter", "example-distribution", d)
-                @test read_distribution(api, "parameter", "example-distribution") == d
+                write_distribution(api, "parameter-write", "example-distribution", d)
+                @test read_distribution(api, "parameter-write", "example-distribution") == d
             end
 
-            write_samples(api, "parameter", "example-samples", [9, 10, 11])
-            @test read_sample(api, "parameter", "example-samples") ∈ [9, 10, 11]
+            @test_throws Exception read_estimate(api, "parameter-write", "example-samples")
+            write_samples(api, "parameter-write", "example-samples", [9, 10, 11])
+            @test read_sample(api, "parameter-write", "example-samples") ∈ [9, 10, 11]
 
+            @test_throws Exception read_estimate(api, "object-write", "example-table")
             df = DataFrame(:a => [9, 10], :b => [11, 12])
-            write_table(api, "object", "example-table", df)
-            @test_broken read_table(api, "object", "example-table") == df
+            write_table(api, "object-write", "example-table", df)
```

Comment:
Is the renaming functionally necessary or just for clarity?

Reply:
It means the write will go to the file defined by `object-write/example-table` in `metadata.yaml` (which doesn't exist, so it will create it). I think it's a bit clearer to separate them, also the fix in data_pipeline_api for over-writing isn't released yet.

Reply:
OK. What i was going towards is whether an additional test is needed to test whether changes are expected (or not) if it already exists, i.e. would the `write` function indeed do an update if the file already exists. If the fix is coming in the pipeline no worries. I thought in that case it was a deliberate choice to not include write functions as part oft he same session. Not sure though.

Reply:
Ah ok, yeah I just checked and that still fails:
```julia
write_estimate(api, "parameter-read", "example-estimate", 99.0)
@test read_estimate(api, "parameter-read", "example-estimate") == 99.0
...
Write API: Test Failed at /Users/sean/github/SimulationData.jl/test/pyapi.jl:30
  Expression: read_estimate(api, "parameter-read", "example-estimate") == 99.0
   Evaluated: 1.0 == 99.0
```
I think the intention is to write stuff out in the same api session, you can write to arbitrary files, but to read them back in with the api you need corresponding entries in `metadata.yaml` (which are not generated automatically AFAIK). Here I've just put those entries in manually so we can easily check the writes succeed.

Reply:
I think this PR may fix the updating case: https://github.com/ScottishCovidResponse/data_pipeline_api/pull/60

We can try it when there is a new `data_pipeline_api` release

Reply:
Cool makes sense. Hope that the version is bumped there otherwise we wont' pick it up (doesn't look like the version is incremented, maybe they tend to do that in subsequent MRs)

Reply:
Speaking of, we probably need a version bump for this too.